scripts/pylib/twister/twisterlib/platform.py

```diff
@@ -50,41 +54,55 @@ def __init__(self):
         self.filter_data = dict()
         self.uart = ""
         self.resc = ""
+        self.qualifier = None
+
+    def load(self, board, target, aliases, data=None):
+        """Load the platform data from the board data and target data
+        board: the board object as per the zephyr build system
+        target: the target name of the board as per the zephyr build system
+        aliases: list of aliases for the target
+        data: the data from the twister.yaml file for the target
+        """
+        self.name = target
+        self.aliases = aliases
 
-    def load(self, platform_file):
-        scp = TwisterConfigParser(platform_file, self.platform_schema)
-        scp.load()
-        data = scp.data
+        v = data.get("variants", {})
```

Comment:
data is now used before checking if is not None:
```
>>> data = None
>>> data.get("variants", {})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```
initialize data before use:
```
    if data is None:
        data = {}
```
or do not use a default value in funcion parameters
(see @PerMac comment https://github.com/zephyrproject-rtos/zephyr/pull/77250#discussion_r1804821953)

Reply:
addressed